c++/src/kj/compat/http.c++

```diff
@@ -2120,37 +2168,56 @@ public:
 
 class HttpFixedLengthEntityWriter final: public kj::AsyncOutputStream {
 public:
-  HttpFixedLengthEntityWriter(HttpOutputStream& inner, uint64_t length)
+  HttpFixedLengthEntityWriter(ResetableRef<HttpOutputStream>& inner, uint64_t length)
       : inner(inner), length(length) {
-    if (length == 0) inner.finishBody();
+    if (length == 0) KJ_ASSERT_NONNULL(inner.tryGet()).finishBody();
   }
   ~HttpFixedLengthEntityWriter() noexcept(false) {
-    if (length > 0 || inner.isWriteInProgress()) {
-      inner.abortBody();
+    auto& i = KJ_UNWRAP_OR(inner.tryGet(), {
```

Comment:
I think all of these have to be `[&] { ... }()` to keep MSVC happy. :(

Reply:
@jasnell apparently I have led you astray: https://github.com/capnproto/capnproto/blob/master/c%2B%2B/src/kj/common.h#L1346-L1349
> // Unfortunately, these macros seem impossible to express without using GCC's non-standard
> // "statement expressions" extension. IIFEs don't do the trick here because a lambda cannot
> // return out of the parent scope. These macros should therefore only be used in projects that
> // target GCC or GCC-compatible compilers.
:(